### DISCUSSION TITLE
SOLVED: Can't flash on Wyze v3

### DISCUSSION THREAD

**Discussion by reidprichard** [gtxaspec/wz_mini_hacks#476] [General]
SOLVED: `git config core.autocrlf` was set to true, automatically converting all files in the repo to Windows style line-endings. Sigh.

I had this working some months ago, but ended up going to Wyze's RTSP firmware. I've decided to circle back, and for the life of me I can't get it working. When I first booted the camera with the wz_mini_hacks files on the SD card, everything went perfectly - the audio prompts played and the RTSP feed became available. I decided to tweak some settings, so I removed the SD card, edited the .conf file, reinserted the card, and plugged the camera back in. Nothing. The camera just sits and intermittently makes clicking noises with the red LED slowly flashing, and it is not reachable in the Wyze app.

Things I've tried:
- Two different SD cards (8GB and 16GB)
- Reformatting the SD cards with the Windows built-in tool (FAT32, default allocation unit size)
- Reformatting the SD card with the formatting tool from sdcard.org (FAT32, 32kB allocation unit size) this has been the trick in the past when flashing Wyze cams)
- Flashing on a new Wyze cam fresh out of the box (my existing ones have RTSP firmware, plus their network access is restricted)
- Re-cloning the repo and copying the files over without any modification
- Boot the camera with no SD card, wait a bit, then unplug it and boot it with the SD card installed
- My final attempt combined everything: 
  - Reformatted the 8GB card
  - Copied over the unmodified SD_ROOT files
  - Booted the cam with no SD card and waited ~5mins
  - Rebooted the cam again with no SD card and waited ~5mins
  - Unplugged the cam, inserted the SD card, plugged it in, and waited 10mins

How do I know it hasn't worked? Well, to be honest I'm not 100% certain, but here are my indications:
- No text-to-speech when the camera boots
- If I enable RTSP in the .conf file, I cannot access the stream via VLC
- SSH times out (99% sure I correctly added my public key)
- My WAP shows the hostname as "WyzeCam", not "WCV3"

The fact that I've had this working before (and that it worked the first time I tried it today) is driving me NUTS. I recall having some difficulties when I got it working months ago, and I'm really wishing I'd written down the secret (or maybe I did and forgot). Anyone have suggestions?

My camera is currently on firmware 4.36.8.15, which should be supported. However, I'm going to try flashing 4.36.9.139 to see if that makes a difference. Edit: 4.36.9.139 flashed successfully, but wz_mini_hacks still doesn't work.

**Comment by reidprichard** [comment]
I flashed another one of my cameras from the RTSP firmware to 4.36.8.15. Like the first, it just flashes red when I try to boot with wz_mini. I assume it is stuck somewhere in the boot process, because it does not show up as connected to my WiFi. 

I don't understand: how does a brand new camera - with exactly the recommended firmware version and completely default settings - fail to work? It doesn't even log anything to the SD card as far as I see, so I have no clue what I can even begin to try.

**Comment by reidprichard** [comment]
Wait - it's working! I found an old version of wz_mini from when I originally got this working back in December. Those may have been the files I used the first time I tried it today - it's been a bit pf a blur, but I think I initially tried those and it was just audio, no video, so I re-downloaded the repo.

I'll try to figure out why it's working and post back.

Edit: 
- I've verified it works with both cameras and with both SD cards, so those can be ruled out. 

Okay, I cannot for the life of me figure out what's going on here. When I compare the working directory with a freshly-cloned one, I get a list of 141 different files (331 were the same):
```
['wz_mini.conf', 'car_control.sh', 'cmd', 'factorycheck', 'gather_wz_logs.sh', 'ip6tables', 'iptables', 'iwconfig.sh', 'iwlist', 'neofetch', 'rtmp-stream.sh', 'spotlight_ctl.sh', 'upgrade-run.sh', 'wpa_cli.sh', 'wz_mini_edit.sh', 'dnsmasq.conf', 'fstab', 'hosts', 'inittab', 'passwd', 'profile', 'rc.common', 'rc.local', 'resolv.conf', 'resolv.dnsmasq', 'shadow', 'uvc.config', 'uvc_jxf22.config', 'uvc_jxf23.config', 'uvc_v2.config', 'wz_mini.conf.dist', 'alsa.conf', 'root', 'S01bind', 'S02tmpfs', 'S03busybox', 'S04model', 'S05mount', 'S06car', 'S07inject', 'S08passwd', 'S09firstboot', 'S10kmod', 'S11configbackup', 'S12ssh', 'S14terminfo', 'S15fwupdate', 'S16factorycheck', 'S18ntp', 'S20initsetup', 'S21killmisc', 'S30patchicamera', 'wz_init.sh', 'S01wlanhw', 'S03ipv6', 'S04wireguard', 'S08hostname', 'S09dnsmasq', 'S10httpd', 'S12remoteaccessory', 'S14storemac', 'S15v4l2rtspserver', 'S16rtmp', 'K01network', 'K02webcam', 'K03rcd', 'K15logging', 'S01swap', 'S03ethernet', 'S04usbdirect', 'S05rndis', 'S06networkalt', 'S07bonding', 'S09coredump', 'S10firmware', 'S13mp4write', 'S14nightdrop', 'S17motor', 'S18fps', 'S20cron', 'S21syslog', 'authorized_keys', 'modules.alias', 'modules.builtin', 'modules.dep', 'modules.devname', 'modules.order', 'modules.softdep', 'modules.symbols', 'modules.alias', 'modules.builtin', 'modules.dep', 'modules.order', 'modules.softdep', 'modules.symbols', 'atbm_txpwer_dcxo_cfg.txt', 'app.ver', 'iCamera', 'iCamera-dbg', 'imp_helper.sh', 'netmon.sh', 'patch_icamera.sh', 'restart_wlan0.sh', 'watch_up.sh', 'cam.js', 'config.js', 'index.html', 'multicam.html', 'car.css', 'car.html', 'car.js', 'BONDING_ENABLED.md', 'DISABLE_FW_UPGRADE.md', 'ENABLE_MP4_WRITE.md', 'ENABLE_USB_DIRECT.md', 'ENABLE_USB_ETH.md', 'ENABLE_USB_STORAGE.md', 'ENABLE_WIREGUARD.md', 'REMOTE_SPOTLIGHT.md', 'RTMP_STREAM_ENABLED.md', 'RTSP_HI_RES_ENABLED.md', 'WEB_CAM_ENABLE.md', 'cam-NIGHT_LED_flag.md', 'cam-bitRate.md', 'cam-drawBoxSwitch.md', 'cam-fps.md', 'cam-indicator.md', 'cam-nightVision.md', 'cam-night_cut_thr.md', 'cam-res.md', 'cam-timezone.md', 'cam-values.txt', 'cam-watermark_flag.md', 'cam.cgi', 'car.sh', 'config.cgi', 'config.css', 'diagnostics.cgi', 'jpeg.cgi', 'numerics.txt', 'shared.cgi']
```

**Comment by reidprichard** [comment]
Love it. Turns out git automatically changes line endings to CRLF on Windows. FML, there's 4 hours of troubleshooting for nothing.

For anyone who cares, you can check this setting with `git config core.autocrlf`. It's *supposed* to be false by default, but I am 110% certain I have never changed mine. Who knows, maybe some project I've used changed it without my knowledge (if that's even possible).

**Reply by Sanfam** [reply]
I'm going to chime in with an affirmative "Check this first."  Ironically, I normally edit files with whitespace displayed, but _do not_ display line returns.  In my scenario, my machine (Windows 10) is running a git install care of another application and I had opted to both download the files initially, then later clone the repo contents into another location and work from those.  I was only ever able to make things work on the first boot, where my source files were the initial set I had direct downloaded, but all subsequent edits were pulled from a snipped fetched from the repo. 

Kudos, @reidprichard !

**Reply by Bob-AOL** [reply]
Windows plain ext editors will use the Unix (LF) convention when editing a text file originating as such and have been doing so since 2018. Copying/pasting and editing such a file using Notepad (or other plain text Windows editor) will generally result in the Windows (CRLF) convention then being applied. The method being used is normally displayed at the bottom of the editor window.

Way too tl:dr, but there's this...

https://www.aleksandrhovhannisyan.com/blog/crlf-vs-lf-normalizing-line-endings-in-git/#configuring-line-endings-in-git-with-coreautocrlf

**Reply by Bob-AOL** [reply]
wz_mini.conf straight from the distro opened with Notepad.

![Unix (LF)](https://user-images.githubusercontent.com/110629141/231583083-c3b66ecf-b250-48dd-a6f3-3e7a4439e7bd.jpg)
